src/transformers/generation/utils.py

```diff
@@ -2728,6 +2548,11 @@ def beam_search(
         ['Wie alt bist du?']
         ```"""
         # init values
+        generation_config = (
+            self.generation_config
+            if self.generation_config is not None
+            else GenerationConfig.from_model_config(self.config)
```

Comment:
IMO the better place is to do this in `from_pretrained(...)` and `__init__(...)` of the model instead of here as mentioned in: https://github.com/huggingface/transformers/pull/20388/files#r1033599800